### PR TITLE
Add header link to article-section pattern

### PR DESCRIPTION
**Pull request by davidcmoulton**

### CODE REVIEW

File: assets/sass/patterns/molecules/article-section.scss [hunk 1]
```diff
@@ -26,6 +26,13 @@ $arrow-size: 5;
   @include nospace("bottom");
 }
 
+.article-section__header_link {
+  display: block;
```
Comment (thewilkybarkid) [inline assets/sass/patterns/molecules/article-section.scss]: This leads to a full-width link.

File: assets/sass/patterns/molecules/article-section.scss [hunk 2]
```diff
@@ -26,6 +26,13 @@ $arrow-size: 5;
...
+  font-family: $font-secondary;
+  @include font-size-and-vertical-height(14, $blg-space-small-in-px);
+  @include padding($blg-space-extra-small-in-px, "bottom");
```
Comment (thewilkybarkid) [inline assets/sass/patterns/molecules/article-section.scss]: Increases the height of the link.

File: source/_patterns/01-molecules/components/article-section.mustache [hunk 3]
```diff
@@ -7,6 +7,9 @@
 
   <header class="article-section__header">
     <h{{headingLevel}} class="article-section__header_text">{{{title}}}</h{{headingLevel}}>
+    {{#headerLink}}
+      <a href="{{{url}}}" class="article-section__header_link">{{name}}</a>
```
Comment (thewilkybarkid) [inline source/_patterns/01-molecules/components/article-section.mustache]: `url` shouldn't be escaped and `name` should be?

File: assets/js/components/ArticleSection.js [hunk 4]
```diff
@@ -60,11 +60,11 @@ module.exports = class ArticleSection {
 
     if ($elm.dataset.initialState === 'closed') {
       $elm.classList.add('article-section--collapsed');
-      $headerLink.classList.add('article-section__header_link--closed');
+      $headerLink.classList.add('article-section__toggle--closed');
```
Comment (thewilkybarkid) [inline assets/js/components/ArticleSection.js]: `$headerLink` needs renaming too.

File: assets/sass/patterns/molecules/article-section.scss (hunk 2 above)
Comment (davidcmoulton) [reply assets/sass/patterns/molecules/article-section.scss]: Approved by Chris H.

File: source/_patterns/01-molecules/components/article-section.mustache (hunk 3 above)
Comment (davidcmoulton) [reply source/_patterns/01-molecules/components/article-section.mustache]: That's what this does. Assumes no HTML in `name`.

File: assets/sass/patterns/molecules/article-section.scss (hunk 2 above)
Comment (thewilkybarkid) [reply assets/sass/patterns/molecules/article-section.scss]: Ok, just odd that's it's not both directions.

File: source/_patterns/01-molecules/components/article-section.mustache (hunk 3 above)
Comment (thewilkybarkid) [reply source/_patterns/01-molecules/components/article-section.mustache]: Ok if `name` is mean to be a plain string, but `url` needs to be too.

Comment (davidcmoulton) [reply source/_patterns/01-molecules/components/article-section.mustache]: 👍  Will escape `url` too.

**Pull request merged**